src/backend/egl/mod.rs

```diff
@@ -472,6 +472,46 @@ impl EGLDisplay {
             Err(BufferAccessError::ContextLost)
         }
     }
+
+    /// Try to receive the dimensions of a given [`WlBuffer`].
+    ///
+    /// In case the buffer is not managed by EGL (but e.g. the [`wayland::shm` module](::wayland::shm))
+    /// a [`BufferAccessError::NotManaged`](::backend::egl::BufferAccessError::NotManaged) is returned with the original buffer
+    /// to render it another way.
+    pub fn egl_buffer_dimensions(
+        &self,
+        buffer: WlBuffer,
+    ) -> ::std::result::Result<(i32, i32), BufferAccessError> {
```

Comment:
I think this function can just be `fn(&self, &WlBuffer) -> Option<(i32, i32)>` tbh. The reuse of `BufferAccessError` here makes thing more complicated than they need to be.